src/RSocketServer.h

```diff
@@ -85,6 +85,12 @@ class RSocketServer {
 
   void shutdownAndWait();
 
+  /**
+   * Gets the port the ConnectionAcceptor is listening on.  Returns folly::none
+   * if this server is not listening on a port.
+   */
+  folly::Optional<uint16_t> listeningPort() const;
```

Comment:
I'm not sure I understand.  Why would HTTP and Thrift not be able to expose the port they're listening on?  This has nothing to do with TCP port 0, this is just "what port is my server listening on".